protocol/wakusync/response.go

```diff
@@ -0,0 +1,25 @@
+package wakusync
+
+import (
+	"encoding/json"
+
+	"github.com/status-im/status-go/protocol/protobuf"
+)
+
+type SyncFromWakuResponse struct {
+	SyncingProgressDetails map[string]protobuf.BackupDataDetails // key represents the data/section backup details refer to
+	Profile                *SyncProfile
+}
+
+func (sfwr *SyncFromWakuResponse) MarshalJSON() ([]byte, error) {
```

Comment:
So,
I think the reason we are doing this marshaling etc, is because we want to convert the protobuf to our own struct:

```
type SyncSectionDetails struct {
	DataNumber  uint32 `json:"dataNumber,omitempty"`
	TotalNumber uint32 `json:"totalNumber,omitempty"`
}
```
```
message BackupDataDetails {
  uint32 data_number = 1;
  uint32 total_number = 2;
}
```

But otherwise the two are equivalent.
I think that's fine, we should not have protobuf moving around, although we do that sometimes but likely we should not.

If we are doing so though, the best place to do the conversion is not at marshal time, can we not just store it as `SyncSectionDetails` when we add it? That way we don't have to override the MarshalJSON function.
Also why two separate files? it seems like this could be a single one

Reply:
@cammellos I wanted not to keep everything in a single file, but have more shorter files, that way I find it easier for maintaining later. What was my idea, so the structure is like this:
```
protocol/wakusync/
    - profile_response.go
    - progress_response.go
    - response.go
    - settings_response.go // will be added later
    - communities_response.go // will be added later
    - ...
```
latter we will add here `settings_response.go`, `communities_response.go` and so on when we have the full list what else need to be backed up.

Now `response.go` is the main file there, and describes the response which will be sent via signal and it looks like this:
```golang
type SyncFromWakuResponse struct {
	SyncingProgressDetails map[string]protobuf.BackupDataDetails
	Profile                *SyncProfile
	Settings               *SyncSettings // will be added later
	Communities            *SyncCommunities // will be added later
}
```
That way I wanted to introduce a bit of consistency in responses, that all signals which are sending data being fetched from waku will have the same form, for example if we're sending data that profile info is fetched that struct will have only `Profile` which carrying info what is changed and `SyncingProgressDetails` describing the current progress of fetching from waku.

Responses would look like:
- when we receive profile 
```
SyncFromWakuResponse { 
    Profile: SyncProfile, 
    SyncingProgressDetails:  map[string]BackupDataDetails 
}
```
- when we receive settings 
```
SyncFromWakuResponse { 
    Settings: SyncSettings, 
    SyncingProgressDetails:  map[string]BackupDataDetails 
}
```
and so on, but that was my general idea, to have clean state.